waddrmgr/scoped_manager.go

```diff
@@ -2289,14 +2305,8 @@ func (s *ScopedKeyManager) importScriptAddress(ns walletdb.ReadWriteBucket,
 		if err != nil {
 			return nil, maybeConvertDbError(err)
 		}
-	}
 
-	// Now that the database has been updated, update the start block in
-	// memory too if needed.
-	if updateStartBlock {
-		s.rootManager.mtx.Lock()
-		s.rootManager.syncState.startBlock = *bs
-		s.rootManager.mtx.Unlock()
+		startBlockModified = true
```

Comment:
Why not put the defer here instead of using the startBlockModified variable?

Reply:
Because the defer would execute before releasing the scoped manager lock. The defer unlock should be executed before taking the manager lock.